src/libraries/System.Text.Json/src/System/ReflectionExtensions.cs

```diff
@@ -42,6 +43,39 @@ public static bool IsAssignableFromInternal(this Type type, Type from)
         private static bool HasJsonConstructorAttribute(ConstructorInfo constructorInfo)
             => constructorInfo.GetCustomAttribute<JsonConstructorAttribute>() != null;
 
+        public static bool HasRequiredMemberAttribute(this ICustomAttributeProvider memberInfo)
+        {
+#if NET7_0_OR_GREATER
+            return memberInfo.IsDefined(typeof(RequiredMemberAttribute), inherit: true);
```

Comment:
I can see why this would need to be `inherit: true` on the type level, but should the property be inherited on the property. I think this largely depends on the semantics of C#, i.e. can you have a `required virtual` property that gets overridden by a non-required variant?

Reply:
I'm getting error when I try to do that: `CS9030	'RequiredKeywordTests.Bar.FooProp' must be required because it overrides required member 'RequiredKeywordTests.Foo.FooProp'`

I believe spec also mentioned to inherit

Reply:
Sounds good then.